libmcount/dynamic.c

```diff
@@ -357,22 +359,75 @@ static bool match_pattern_list(struct list_head *patterns,
 	return ret;
 }
 
-static int do_dynamic_update(struct symtabs *symtabs, char *patch_funcs,
-			     enum uftrace_pattern_type ptype)
+static void patch_func_matched(struct list_head *patterns,
+			       struct mcount_dynamic_info *mdi,
+			       struct uftrace_mmap *map)
 {
-	struct uftrace_mmap *map;
+	bool found = false;
 	struct symtab *symtab;
-	struct strv funcs = STRV_INIT;
-	char *def_mod;
-	char *name;
-	int j;
+	bool csu_skip;
+	unsigned i, k;
+	struct sym *sym;
 	/* skip special startup (csu) functions */
 	const char *csu_skip_syms[] = {
 		"_start",
 		"__libc_csu_init",
 		"__libc_csu_fini",
 	};
-	LIST_HEAD(patterns);
+
+	symtab = &map->mod->symtab;
+
+	for (i = 0; i < symtab->nr_sym; i++) {
+		sym = &symtab->sym[i];
+
+		csu_skip = false;
+		for (k = 0; k < ARRAY_SIZE(csu_skip_syms); k++) {
+			if (!strcmp(sym->name, csu_skip_syms[k])) {
+				csu_skip = true;
+				break;
+			}
+		}
+		if (csu_skip)
+			continue;
+
+		if (sym->type != ST_LOCAL_FUNC &&
+		    sym->type != ST_GLOBAL_FUNC)
+			continue;
+
+		if (!match_pattern_list(patterns, map, sym->name)) {
+			if (mcount_unpatch_func(mdi, sym, &disasm) == 0)
+				stats.unpatch++;
+			continue;
+		}
+
+		found = true;
+		switch (mcount_patch_func(mdi, sym, &disasm, min_size)) {
+			case INSTRUMENT_FAILED:
+				stats.failed++;
+				break;
+			case INSTRUMENT_SKIPPED:
+				stats.skipped++;
+				break;
+			case INSTRUMENT_SUCCESS:
+			default:
+				break;
+		}
+		stats.total++;
+	}
+
+	if (!found)
+		stats.nomatch++;
+
```

Comment:
Please remove this unnecessary blank line.

Reply:
I swear to God, I modified it before... At least it was fixed now.